ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -1026,6 +1026,17 @@ func (r *RayClusterReconciler) reconcileAutoscalerServiceAccount(instance *rayio
 			return err
 		}
 
+		// If users specify ServiceAccountName for the head Pod, they need to create a ServiceAccount themselves.
+		// However, if KubeRay creates a ServiceAccount for users, the autoscaler may encounter permission issues during
+		// zero-downtime rolling updates when RayService is performed. See https://github.com/ray-project/kuberay/issues/1123
+		// for more details.
+		if instance.Spec.HeadGroupSpec.Template.Spec.ServiceAccountName == namespacedName.Name {
+			r.Log.Error(err, fmt.Sprintf(
+				"If users specify ServiceAccountName for the head Pod, they need to create a ServiceAccount themselves."+
```

Comment:
Maybe we can also remind the user to ensure at least the following permissions are granted: ["get", "list", "watch", "patch"](https://github.com/ray-project/kuberay/blob/fe29409358e911c2eecb6da6b0a530479f911567/ray-operator/controllers/ray/common/rbac.go#L29) on the pods. These permissions are necessary for the autoscaler to adjust the number of replicas based on the workload.

Reply:
The permissions are defined by Role which is created by KubeRay automatically. Do we still need to add the comment?

Reply:
oh! I see, kuberay will automatically create `role` and `rolebinding` for the user created ServiceAccountName when `reconcileAutoscalerRole` and `reconcileAutoscalerRoleBinding`